docs/keto/guides/rbac.mdx

```diff
@@ -2,6 +2,201 @@
+where one often encounters roles such as "administrator", "moderator", and so
+on.
+
+In **Hierarchical Role Based Access Control (HRBAC)** roles can inherit
+permissions from other roles. The "administrator" role, for example, could
+inherit all permissions from the "moderator" role. This reduces duplication and
+management complexity around defining privileges.
+
+Let's come back to Alice, Bob, Peter, and blog posts and the matrix from the ACL
+example. This time we model the access rights using (H)RBAC and the roles
+`reader`, `author`, and `admin`:
+
+The role `admin` inherits all privileges from `author`, which inherits from
+`reader`. Only `Alice` (or rather her role `admin`) can delete blog posts,
+whereas `author` can create and modify blog posts.
+
+(H)RBAC is everywhere. If you ever installed a forum software such as
+[phpBB](https://www.phpbb.com/support/docs/en/3.1/ug/adminguide/permissions_roles/)
+or [Wordpress](https://codex.wordpress.org/Roles_and_Capabilities), you have
+definitely encountered ACL, (H)RBAC, or both.
+
+(H)RBAC reduces management complexity and overhead with large user bases.
+Sometimes however, even (H)RBAC is not enough. An example is when you need to
+express ownership (e.g. `Bob` can only modify his own blog posts), have
+attributes (e.g. `Bob` works in department `blog`), or in multi-tenant
+environments.
+
+**Benefits:**
+
+- Reduces management complexity when many identities share similar permissions.
+- Role hierarchies can reduce redundancy even further.
+- Is well established and easily understood by many developers as it is a
+  de-facto standard for web applications.
+
+**Shortcomings:**
+
+- Has no concept of context:
+  - There is no concept of ownership: _Dan is the author of article "Hello
+    World" and is thus allowed to update it_.
+  - There is no concept of environment: _Dan is allowed to access accounting
+    services when the request comes from IP 10.0.0.3_.
+  - There is no concept of tenants: _Dan is allowed to access resources on the
+    "dan's test" tenant_.
+
+## Using RBAC with Ory Keto
+
+Let's assume that we are building a reporting application and need to have three
+groups of users with different access levels. We have the following group of
+reports in our application.
+
+- Financial performance reports
+- Marketing performance reports
+- Community performance reports
+
+We need to have three groups, `finance`, `marketing`, `community`. Also, we need
+to have two namespaces: `reports` to manage access control and `groups` to add
+users to this group
+
+Let's add namespaces to Keto config.
+[here](https://www.ory.sh/docs/keto/reference/configuration)
+
+```yaml
+# ...
+namespaces:
+  - id: 0
+    name: groups
+  - id: 1
+    name: reports
+# ...
+```
+
+We can have two types of permission to access reports for granularity. Let's
+assume that we need `edit` and `view` access to the reports.
+
+```keto-relation-tuples
```

Comment:
We have a custom prism language for `keto-relation-tuples` :wink: so syntax highlighting _should_ work. Currently the colors are somehow broken, but the tokens are correctly recognized.